cclib/bridge/cclib2ase.py

```diff
@@ -0,0 +1,243 @@
+        if hasattr(ccdata, "scfenergies"):
+            properties.update({"energy": ccdata.scfenergies[i]})
+        if hasattr(ccdata, "grads"):
+            try:
+                properties.update(
+                    {"forces": -ccdata.grads[i] * units.Hartree / units.Bohr}
+                )
+            except IndexError:
+                pass
+
+        if i == len(ccdata.atomcoords) - 1:  # last geometry
+            if hasattr(ccdata, "moments"):
+                properties.update({"dipole": ccdata.moments[1] * units.Bohr})
+            if hasattr(ccdata, "free_energy"):
+                properties.update(
+                    {"free_energy": ccdata.freeenergy * units.Hartree}
+                )
+
+        traj.write(atoms, **properties)
+
+
+def read_trajectory(filename):
+    """Read an ASE Trajectory object and return a ccData object.
+
+    The returned object has everything write_trajectory writes, plus natom,
+    charge, mult and temperature.
+
+    The following properties are taken from the last frame: atomnos,
+    atomcharges, atomspins, atommasses, moments, freeenergy and temperature.
+    charge, mult and natom also represent the last frame, since they depend on
+    other propertes read from the last frame.
+
+    Bear in mind that ASE calculates temperature from the kinetic energy, so
+    anything "static" (which includes anything cclib parses) will have zero
+    temperature.
+
+    Inputs:
+        filename - path to traj file to be read.
+    """
+    _check_ase(_found_ase)
+    attributes = {"atomcoords": [], "scfenergies": [], "grads": []}
+
+    for atoms in Trajectory(filename, "r"):
+        ccdata = makecclib(atoms)
+        attributes["atomcoords"].append(ccdata.atomcoords[-1])
+        if hasattr(ccdata, "scfenergies"):
+            attributes["scfenergies"].append(ccdata.scfenergies[-1])
+        if hasattr(ccdata, "grads"):
+            attributes["grads"].append(ccdata.grads[-1])
+
+    # ccdata is now last frame
+    attributes["atomnos"] = ccdata.atomnos
+    attributes["atomcharges"] = ccdata.atomcharges
+    attributes["atomspins"] = ccdata.atomspins
+    attributes["atommasses"] = ccdata.atommasses
+
+    if hasattr(ccdata, "moments"):
+        attributes["moments"] = ccdata.moments
+    if hasattr(ccdata, "freeenergy"):
+        attributes["freeenergy"] = ccdata.freeenergy
+
+    # remove if empty
+    if not attributes["scfenergies"]:
+        del attributes["scfenergies"]
+    if not attributes["grads"]:
+        del attributes["grads"]
+
+    # extra stuff we can't write in write_trajectory
+    attributes["temperature"] = ccdata.temperature
+    attributes["charge"] = ccdata.charge
+    attributes["mult"] = ccdata.mult
+    attributes["natom"] = ccdata.natom
+
+    return ccData(attributes)
+
+
+def makecclib(atoms, popname="mulliken"):
+    """Create cclib attributes and return a ccData from an ASE Atoms object.
+
+    Available potential energy is assumed to be from SCF (see

```

Comment:
Can you mention forces/gradients here as well?

Reply:
Sure, done.